src/main.rs

```diff
@@ -43,30 +48,31 @@ fn print_connector_info(gpu: &Card, resources: &ResourceHandles) {
             .get_connector(*connector_handle, force_probe)
             .expect("Failed to get GPU connector info");
 
-        let connector_interface = connector.interface().as_str();
-        let interface_id = connector.interface_id();
+        let name = get_connector_name(&connector);
 
         let connection_state = match connector.state() {
             ConnectorState::Connected => "✅",
             ConnectorState::Disconnected => "❌",
             ConnectorState::Unknown => "❔",
         };
 
-        println!("\t{connector_interface}-{interface_id}, Connected={connection_state}");
+        println!("\t{name}, Connected={connection_state}");
     }
 }
 
-fn first_connected_connector(gpu: &Card) -> Option<ConnectorHandle> {
+fn get_connected_connector(gpu: &Card, name: &Option<String>) -> Option<ConnectorHandle> {
     gpu.resource_handles()
         .expect("Failed to get GPU resource handles")
         .connectors
         .iter()
         .find(|connector_handle| {
             let force_probe = false;
-            gpu.get_connector(**connector_handle, force_probe)
-                .expect("Failed to get GPU connector info")
-                .state()
-                == ConnectorState::Connected
+            let info = gpu
+                .get_connector(**connector_handle, force_probe)
+                .expect("Failed to get GPU connector info");
+            let name_matches =
+                if let Some(name) = name { name == &get_connector_name(&info) } else { true };
+            name_matches && info.state() == ConnectorState::Connected
```

Comment:
nit: maybe this is less convoluted? (Not a strong opinion)

```suggestion
            if info.state() != ConnectorState::Connected {
                false
            } else if let Some(name) = name {
                name == &get_connector_name(&info)
            } else {
                true
            }
```

Reply:
Yeah, I agree. Not sure which way is better, so I have rewritten it in a third way :smile: 

I'll let you judge the new version but I'd leave  it be for the moment, I suspect we'll end up rewriting this stuff again anyway as we learn more.

Reply:
I'm happy with the end result. I enjoyed the code-golf everybody got to play on this little example!